src/key_store/mod.rs

```diff
@@ -0,0 +1,88 @@
+use std::collections::HashMap;
+
+use crate::{
+    ciphersuite::{CiphersuiteName, SignaturePublicKey, SignatureScheme},
+    credentials::{Credential, CredentialBundle, CredentialType},
+    extensions::Extension,
+    key_packages::{KeyPackage, KeyPackageBundle},
+};
+
+pub mod errors;
+
+pub use errors::KeyStoreError;
+
+#[derive(Debug, Default)]
+/// The `KeyStore` contains private key material of `Credentials` and
+/// `KeyPackage`s.
+pub struct KeyStore {
+    // Map from signature public keys to credential bundles
+    credentials: HashMap<SignaturePublicKey, CredentialBundle>,
+    key_packages: HashMap<Vec<u8>, KeyPackageBundle>,
+}
+
+impl KeyStore {
+    /// Retrieve a `KeyPackageBundle` from the key store given the hash of the
+    /// corresponding `KeyPackage`. Returns an error if no `KeyPackageBundle`
+    /// can be found corresponding to the given `KeyPackage` hash. TODO: This is
+    /// not in use yet, because the groups are not yet refactored to use the key
+    /// store for KeyPackageBundles.
+    pub(crate) fn _key_package_bundle(
+        &self,
+        kp_hash: &[u8],
+    ) -> Result<&KeyPackageBundle, KeyStoreError> {
```

Comment:
I'm thinking this should return the value and not just a reference, otherwise we might keep around key package bundles even though they are no longer needed (which is bad for FS).

Reply:
Done!